client/tests/integration/roles.rs

```diff
@@ -141,16 +133,10 @@ fn register_metadata_role() -> Result<()> {
     test_client.submit_blocking(register_bob)?;
 
     let role_id = iroha_data_model::role::Id::new("USER_METADATA_ACCESS".parse::<Name>()?);
-    let mut permissions = Permissions::new();
-    let mut params = BTreeMap::new();
-    params.insert(Name::from_str("account_id")?, bob_id.into());
-    permissions.insert(
-        PermissionToken::new(CAN_SET_KEY_VALUE_IN_USER_METADATA.clone())
-            .with_params(params.clone()),
-    );
-    permissions.insert(
-        PermissionToken::new(CAN_REMOVE_KEY_VALUE_IN_USER_METADATA.clone()).with_params(params),
-    );
+    let permissions: Vec<PredefinedPermissionToken> = vec![
```

Comment:
This is the only place I see `PredefinedPermissionToken` is somewhat useful used, i.e. when creating a new `Role`. However, it will not be possible to mix predefined and generic tokens in the `Role::new` constructor in which case you should make a conversion beforehand of predefined tokens into `Vec<PermissionToken>`. I would consider removing the `PredefinedPermissionToken` enumeration because I don't see much value in it.

Additionally, I would consider making a `PermissionTokenTrait`(name is placeholder) and have `Role::new` have the following signature:
```rs
    pub fn new(
        id: <Self as Identifiable>::Id,
        permissions: impl IntoIterator<Item = Box<dyn PermissionTokenTrait>>
    ) -> <Self as Identifiable>::RegisteredWith {
```

or what I think would be even better is to remove the `PredefinedPermissionToken` and make a builder for `Role`s which would take only one permission at a time

Reply:
I think the enum can be used by clients - you can convert generic token into it and cover all predefined cases.

I don't think `dyn PermissionTokenTrait` will do much good, since you'll have to box your tokens instead of `into`ing them, which doesn't add much convenience and adds inderection. Builder sounds like a good idea though.

Reply:
sounds good